### ISSUE TITLE
canonical URL is incorrect

### ISSUE THREAD

**Issue by discoform** [wurmlab/sequenceserver-website#1]
## Description

Typically updating the canonical URL should occur during the build process, but that doesn't seem to be the case here.

## Expected Behavior

Pushing changes or merging PRs to the main repo should start the build process automatically when using GitHub pages. That build process didn't happen with this latest update. Maybe it's a one-time issue with the new repo?

I'll submit a PR shortly that should address the immediate need.

**Comment by yeban** [comment]
> Pushing changes or merging PRs to the main repo should start the build process automatically when using GitHub pages. That build process didn't happen with this latest update. Maybe it's a one-time issue with the new repo?

The website is hosted on a custom server running Nginx instead of on GitHub. There is no build process. This is why we `_site` folder is currently tracked in git - so that any changes are built locally and pushed at once to the server.

However, the setup is proving problematic now. I see the following untracked changes in my local clone of the repository after running `bundle exec jekyll serve`:

```
diff --git a/_site/404.html b/_site/404.html
index 5dfa0ac..eba2000 100644
--- a/_site/404.html
+++ b/_site/404.html
@@ -12,13 +12,13 @@
 <meta property="og:locale" content="en_US" />
 <meta name="description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
 <meta property="og:description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
-<link rel="canonical" href="https://sequenceserver.com/404.html" />
-<meta property="og:url" content="https://sequenceserver.com/404.html" />
+<link rel="canonical" href="http://localhost:4000/404.html" />
+<meta property="og:url" content="http://localhost:4000/404.html" />
 <meta property="og:site_name" content="SequenceServer" />
 <meta name="twitter:card" content="summary" />
 <meta property="twitter:title" content="Error 404" />
 <script type="application/ld+json">
-{"@type":"WebPage","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","headline":"Error 404","url":"https://sequenceserver.com/404.html","@context":"https://schema.org"}</script>
+{"headline":"Error 404","url":"http://localhost:4000/404.html","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","@type":"WebPage","@context":"https://schema.org"}</script>
 <!-- End Jekyll SEO tag -->
 <meta name="author" content="Anurag Priyam">
   <meta name="author" content="Yannick Wurm">
diff --git a/_site/blog/sequence-search/index.html b/_site/blog/sequence-search/index.html
index 8b9fc25..b3d82f0 100644
--- a/_site/blog/sequence-search/index.html
+++ b/_site/blog/sequence-search/index.html
@@ -12,15 +12,15 @@
 <meta property="og:locale" content="en_US" />
 <meta name="description" content="Sequence Search" />
 <meta property="og:description" content="Sequence Search" />
-<link rel="canonical" href="https://sequenceserver.com/blog/sequence-search/" />
-<meta property="og:url" content="https://sequenceserver.com/blog/sequence-search/" />
+<link rel="canonical" href="http://localhost:4000/blog/sequence-search/" />
+<meta property="og:url" content="http://localhost:4000/blog/sequence-search/" />
 <meta property="og:site_name" content="SequenceServer" />
 <meta property="og:type" content="article" />
-<meta property="article:published_time" content="2021-07-24T23:03:32-07:00" />
+<meta property="article:published_time" content="2021-07-24T23:03:32+01:00" />
 <meta name="twitter:card" content="summary" />
 <meta property="twitter:title" content="Sequence Search" />
 <script type="application/ld+json">
-{"@type":"BlogPosting","description":"Sequence Search","headline":"Sequence Search","dateModified":"2021-07-24T23:03:32-07:00","datePublished":"2021-07-24T23:03:32-07:00","mainEntityOfPage":{"@type":"WebPage","@id":"https://sequenceserver.com/blog/sequence-search/"},"url":"https://sequenceserver.com/blog/sequence-search/","@context":"https://schema.org"}</script>
+{"headline":"Sequence Search","dateModified":"2021-07-24T23:03:32+01:00","datePublished":"2021-07-24T23:03:32+01:00","mainEntityOfPage":{"@type":"WebPage","@id":"http://localhost:4000/blog/sequence-search/"},"url":"http://localhost:4000/blog/sequence-search/","description":"Sequence Search","@type":"BlogPosting","@context":"https://schema.org"}</script>
 <!-- End Jekyll SEO tag -->
 <meta name="author" content="Anurag Priyam">
   <meta name="author" content="Yannick Wurm">
diff --git a/_site/doc/index.html b/_site/doc/index.html
index e3463a6..5721830 100644
--- a/_site/doc/index.html
+++ b/_site/doc/index.html
@@ -12,13 +12,13 @@
 <meta property="og:locale" content="en_US" />
 <meta name="description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
 <meta property="og:description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
-<link rel="canonical" href="https://sequenceserver.com/doc/" />
-<meta property="og:url" content="https://sequenceserver.com/doc/" />
+<link rel="canonical" href="http://localhost:4000/doc/" />
+<meta property="og:url" content="http://localhost:4000/doc/" />
 <meta property="og:site_name" content="SequenceServer" />
 <meta name="twitter:card" content="summary" />
 <meta property="twitter:title" content="SequenceServer" />
 <script type="application/ld+json">
-{"@type":"WebPage","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","headline":"SequenceServer","url":"https://sequenceserver.com/doc/","@context":"https://schema.org"}</script>
+{"headline":"SequenceServer","url":"http://localhost:4000/doc/","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","@type":"WebPage","@context":"https://schema.org"}</script>
 <!-- End Jekyll SEO tag -->
 <meta name="author" content="Anurag Priyam">
   <meta name="author" content="Yannick Wurm">
diff --git a/_site/index.html b/_site/index.html
index 0e23cae..6af8a36 100644
--- a/_site/index.html
+++ b/_site/index.html
@@ -12,13 +12,13 @@
 <meta property="og:locale" content="en_US" />
 <meta name="description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
 <meta property="og:description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
-<link rel="canonical" href="https://sequenceserver.com/" />
-<meta property="og:url" content="https://sequenceserver.com/" />
+<link rel="canonical" href="http://localhost:4000/" />
+<meta property="og:url" content="http://localhost:4000/" />
 <meta property="og:site_name" content="SequenceServer" />
 <meta name="twitter:card" content="summary" />
 <meta property="twitter:title" content="SequenceServer" />
 <script type="application/ld+json">
-{"@type":"WebSite","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","headline":"SequenceServer","name":"SequenceServer","url":"https://sequenceserver.com/","@context":"https://schema.org"}</script>
+{"headline":"SequenceServer","url":"http://localhost:4000/","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","@type":"WebSite","name":"SequenceServer","@context":"https://schema.org"}</script>
 <!-- End Jekyll SEO tag -->
 <meta name="author" content="Anurag Priyam">
   <meta name="author" content="Yannick Wurm">
diff --git a/_site/paper/blast-interface-design-and-implementation.html b/_site/paper/blast-interface-design-and-implementation.html
index 752c425..35d353d 100644
--- a/_site/paper/blast-interface-design-and-implementation.html
+++ b/_site/paper/blast-interface-design-and-implementation.html
@@ -12,13 +12,13 @@
 <meta property="og:locale" content="en_US" />
 <meta name="description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
 <meta property="og:description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
-<link rel="canonical" href="https://sequenceserver.com/paper/blast-interface-design-and-implementation.html" />
-<meta property="og:url" content="https://sequenceserver.com/paper/blast-interface-design-and-implementation.html" />
+<link rel="canonical" href="http://localhost:4000/paper/blast-interface-design-and-implementation.html" />
+<meta property="og:url" content="http://localhost:4000/paper/blast-interface-design-and-implementation.html" />
 <meta property="og:site_name" content="SequenceServer" />
 <meta name="twitter:card" content="summary" />
 <meta property="twitter:title" content="Paper Supplementary Information. Sequenceserver: a modern graphical user interface for custom BLAST databases" />
 <script type="application/ld+json">
-{"@type":"WebPage","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","headline":"Paper Supplementary Information. Sequenceserver: a modern graphical user interface for custom BLAST databases","url":"https://sequenceserver.com/paper/blast-interface-design-and-implementation.html","@context":"https://schema.org"}</script>
+{"headline":"Paper Supplementary Information. Sequenceserver: a modern graphical user interface for custom BLAST databases","url":"http://localhost:4000/paper/blast-interface-design-and-implementation.html","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","@type":"WebPage","@context":"https://schema.org"}</script>
 <!-- End Jekyll SEO tag -->
 <meta name="author" content="Anurag Priyam">
   <meta name="author" content="Yannick Wurm">
diff --git a/_site/paper/index.html b/_site/paper/index.html
index 98ebcf0..2dec38e 100644
--- a/_site/paper/index.html
+++ b/_site/paper/index.html
@@ -12,13 +12,13 @@
 <meta property="og:locale" content="en_US" />
 <meta name="description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
 <meta property="og:description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
-<link rel="canonical" href="https://sequenceserver.com/paper/" />
-<meta property="og:url" content="https://sequenceserver.com/paper/" />
+<link rel="canonical" href="http://localhost:4000/paper/" />
+<meta property="og:url" content="http://localhost:4000/paper/" />
 <meta property="og:site_name" content="SequenceServer" />
 <meta name="twitter:card" content="summary" />
 <meta property="twitter:title" content="Sequenceserver: A Modern Graphical User Interface for Custom BLAST Databases" />
 <script type="application/ld+json">
-{"@type":"WebPage","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","headline":"Sequenceserver: A Modern Graphical User Interface for Custom BLAST Databases","url":"https://sequenceserver.com/paper/","@context":"https://schema.org"}</script>
+{"headline":"Sequenceserver: A Modern Graphical User Interface for Custom BLAST Databases","url":"http://localhost:4000/paper/","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","@type":"WebPage","@context":"https://schema.org"}</script>
 <!-- End Jekyll SEO tag -->
 <meta name="author" content="Anurag Priyam">
   <meta name="author" content="Yannick Wurm">
diff --git a/_site/progress-on-BLAST-interface.html b/_site/progress-on-BLAST-interface.html
index 1036b45..58bcbb7 100644
--- a/_site/progress-on-BLAST-interface.html
+++ b/_site/progress-on-BLAST-interface.html
@@ -12,15 +12,15 @@
 <meta property="og:locale" content="en_US" />
 <meta name="description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
 <meta property="og:description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science." />
-<link rel="canonical" href="https://sequenceserver.com/progress-on-BLAST-interface.html" />
-<meta property="og:url" content="https://sequenceserver.com/progress-on-BLAST-interface.html" />
+<link rel="canonical" href="http://localhost:4000/progress-on-BLAST-interface.html" />
+<meta property="og:url" content="http://localhost:4000/progress-on-BLAST-interface.html" />
 <meta property="og:site_name" content="SequenceServer" />
 <meta property="og:type" content="article" />
-<meta property="article:published_time" content="2021-08-01T23:03:32-07:00" />
+<meta property="article:published_time" content="2021-08-01T23:03:32+01:00" />
 <meta name="twitter:card" content="summary" />
 <meta property="twitter:title" content="SequenceServer BLAST release notes" />
 <script type="application/ld+json">
-{"@type":"BlogPosting","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","headline":"SequenceServer BLAST release notes","dateModified":"2021-08-01T23:03:32-07:00","datePublished":"2021-08-01T23:03:32-07:00","mainEntityOfPage":{"@type":"WebPage","@id":"https://sequenceserver.com/progress-on-BLAST-interface.html"},"url":"https://sequenceserver.com/progress-on-BLAST-interface.html","@context":"https://schema.org"}</script>
+{"headline":"SequenceServer BLAST release notes","dateModified":"2021-08-01T23:03:32+01:00","datePublished":"2021-08-01T23:03:32+01:00","mainEntityOfPage":{"@type":"WebPage","@id":"http://localhost:4000/progress-on-BLAST-interface.html"},"url":"http://localhost:4000/progress-on-BLAST-interface.html","description":"Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.","@type":"BlogPosting","@context":"https://schema.org"}</script>
 <!-- End Jekyll SEO tag -->
 <meta name="author" content="Anurag Priyam">
   <meta name="author" content="Yannick Wurm">
```

Is there a way to tell Jekyll to use `sequenceserver.com` instead of `localhost:4000` and to freeze or avoid changing the timestamps because of different timezones?

**Comment by yeban** [comment]
Reopening as I don't think #1 truly fixes the problem - see the above comment.

**Comment by discoform** [comment]
> Is there a way to tell Jekyll to use `sequenceserver.com` instead of `localhost:4000` and to freeze or avoid changing the timestamps because of different timezones?

- From my experience, the way to tell Jekyll to use the desired production URL is to establish it in `_config.yml` (which we do) and to build with PR merge or change. From what I gather, the missing piece here is the build command? 

- Jekyll will use `localhost` during local development and testing so that assets are linked correctly in the local version. I don't _think_ it would be helpful or advisable to use the production url during local development.

- Re: timestamps, I'm not sure. I have some ideas, but will need to read a bit more about it. My initial theory would be to exclude `_site` in my `.gitignore` to avoid conflicts. I don't know for sure that that would solve it, but it might be worth exploring.

### Question
What are the benefits to the custom Nginx hosting solution? Would we consider simplifying that to a standard GitHub pages deploy? If we need something more feature rich we could consider something like Netlify?

**Comment by yeban** [comment]
> What are the benefits to the custom Nginx hosting solution? Would we consider simplifying that to a standard GitHub pages deploy? If we need something more feature rich we could consider something like Netlify?

It is because of the https://sequenceserver.com/cloud page. The source code for that page is in a different, private repo. It is actually a small Sinatra app. Custom Nginx solution allows us to make that Sinatra app available at `/cloud` in a simple manner.

I think we will want to keep the Nginx setup for the time being.

Do you think excluding _site from the repo and building it on the server using a git push hook can solve the problem?

**Comment by discoform** [comment]
> Do you think excluding _site from the repo and building it on the server using a git push hook can solve the problem?


I do think that could (and should) work. I typically exclude the `_site` folder from the repo if I'm deploying with Netlify. It only requires that I set a build command (to `bundle exec jekyll build`, or simply `jekyll build`) and set a publish directory (to `_site`)

In my case(s) it's set up in the Netlify admin UI, but I'm not sure what kind of options you might have in the custom solution. I looked around a bit today and it there do seem to be some Nginx examples out there. I can dig in a bit more in the morning if needed.
